add article regarding cpu power save (#38591)
ydb-platform/ydb: ydb/docs/en/core/devops/concepts/cpu-production-settings.md

# CPU settings for production {#cpu-production-settings}

## Problem {#problem}

CPU power-saving modes (such as Intel SpeedStep, AMD Cool'n'Quiet, C-states, and P-states) can hurt {{ ydb-short-name }} performance for several reasons:

1. **Unstable CPU frequency** — {{ ydb-short-name }} expects stable CPU throughput for predictable operation.
2. **Core affinity issues** — changing per-core frequency can break load balancing across threads.
3. **Mode-switching latency** — transitions between power-saving states add extra delays.
4. **Unpredictable throughput** — operations may run at different speeds depending on the current CPU mode.

Typical symptoms:

* Unstable query latency.
* Unexplained performance drops.
* Uneven load across cluster nodes.
* Erratic behavior under heavy load.

## Solution {#solution}

### Recommended BIOS/UEFI settings {#bios}

> For {{ ydb-short-name }} servers, disable or tune the following power-saving features:

Author: The intro says "disable or tune the following **power-saving features**", but item 2 (Intel Turbo Boost) is a performance *enhancement* that should be *enabled*, not a power-saving feature. The framing is contradictory.

```suggestion
For {{ ydb-short-name }} servers, configure the following BIOS/UEFI settings:
```

1. **CPU Power Management** — set to **Performance** or **Max Performance**.
2. **Intel Turbo Boost** — enable for maximum performance.
3. **C-states** — disable or limit to C1/C1E.
4. **P-states** — use a fixed maximum frequency.
5. **Intel SpeedStep / AMD Cool'n'Quiet** — disable.
6. **Package C-State Limit** — set to C0 or C1.

### Operating system settings {#os}

> On Linux, additionally consider:

Author: The Ansible deployment role already configures the `performance` governor automatically (via `cpufrequtils`). Readers who deployed with Ansible do not need to run these commands manually. The article should clarify the scope to avoid confusion:

```suggestion
On Linux, for deployments **not** using the [Ansible role](../../deployment-options/ansible/index.md) (which configures this automatically), additionally consider:
```

```bash
# Force the performance cpufreq governor
echo 'performance' | tee /sys/devices/system/cpu/cpu*/cpufreq/scaling_governor

# Pin minimum frequency to maximum
echo $(cat /sys/devices/system/cpu/cpu0/cpufreq/cpuinfo_max_freq) | \
tee /sys/devices/system/cpu/cpu*/cpufreq/scaling_min_freq

# Reduce PCIe ASPM power saving
echo 'performance' > /sys/module/pcie_aspm/parameters/policy
```

### Verifying current settings {#check}

> Useful checks:

Author: The verification section shows commands but does not describe what the correct output should look like. The reader cannot self-verify that the settings were applied correctly.

```suggestion
Useful checks (after applying settings, the first command should output `performance` for every CPU core):
```
